### PR TITLE
create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by ramyaparimi**
This commit has a new contributing.MD file created.

### CODE REVIEW

**Comment by ramyaparimi** [comment]
This is a basic first draft. Please let me know what else can be added/removed for better understanding. 😀

**Review by nasirhm — CHANGES_REQUESTED** [review]
It looks great for a first draft @ramyaparimi, 

I've added some comments for improving it.

Please review and share your thought about them.

**Comment by t0xic0der** [comment]
I don't really think it is right to explore the project's technicalities in the CONTRIBUTING.md file. 

We should restrict the file to describe the methodologies about how folks can contribute by covering the following points.

- Folks can start contributing to the project without even writing a single line of code by testing the project and reviewing the PRs listed at their ends. Should they discover bugs and discrepancies in PRs, they would want to list it by checking first if the same has not been listed before and proceed by explaining the scenario in an understandable manner.
- Folks who want to contribute via code can either comment under a specific issue to express their interest on working on it or contact the Mote modernizing team to know how they can begin contributing. PRs would only be considered for a review if they deal with an issue that the person is assigned to and must be atomic in nature (One PR per issue).
- The document can further explore the nuances about how a fork should be made and be cloned to their local storage. Then, it can describe how PRs should be made exclusively from a non-master branch of their fork as that helps retaining the master branch as a reference to roll back to, should the branch created not work the way it should.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+Your interest in contributing to this project is greatly appreciated. Please know that coding is not the only way to contribute. 
+
+**Here are some ways you can help that doesn't involve coding:**
+
+- Review any active PRs. If the PR needs any changes, please exlpain them in a simple way.
```
Comment (t0xic0der) [inline CONTRIBUTING.md:5]: `sed -i s/exlpain/explain/g`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+- You can test the project to see if it has any bugs. If you come across a bug that has not been addressed before, please explain the scenario in a clear and consice manner.
```
Comment (t0xic0der) [inline CONTRIBUTING.md:6]: `sed -i s/consice/concise/g`

Comment (t0xic0der) [inline CONTRIBUTING.md:6]: Also do explain as to how they can list the bugs. Probably add the link to the issues page and ask them to search through the list of issues first to see if it has not been listed before going ahead and adding them.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+**If coding is your choice of contribution:**
+
+- If you are not sure how to contribute, please contact the Mote modernizing team.
```
Comment (t0xic0der) [inline CONTRIBUTING.md:10]: Please add the link to the matrix channel we have. :smile:

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+**If coding is your choice of contribution:**
+
+- If you are not sure how to contribute, please contact the Mote modernizing team. 
+- If you find an [issue](https://github.com/fedora-infra/mote/issues) that interests you, comment under it so it can be assigned to you. You can only create a PR for an issue you are assigned to.
+- Please keep the PRs atomic (only one PR per issue)
+
+If this is your first time working with git, below are clear instructions on how to proceed.
+
+a. Create your upstream repo:
```
Comment (t0xic0der) [inline CONTRIBUTING.md:16]: "Clone the upstream repo" instead of "Create your upstream repo". I don't know, what do you folks think? :thinking:

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  3. In your terminal, go the the directory you want your repo to be in and run `git clone <GIT URL> -o upstream`. 
+
+b. Create your forked repo:
+
+  1. Go to the upstream repo, click on the fork button on the top right corner. 
+  2. Click on the code dropdown menu, copy the contents in SSH box.
+  3. From your terminal go to the directory where you cloned the upstream repo.
+  4. Run the command `git remote add origin <SSH URL>`.
+  
+c. Before every contribution:
```
Comment (t0xic0der) [inline CONTRIBUTING.md:29]: "Before making a pull request" instead of "Before every contribution" maybe.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+  1. Checkout the `master` branch.
+  2. Get the upstream changes with `git pull upstream master`.
+
+d. Making a contribution:
+
+  1. Create a new branch with `git checkout -b "branch name".
```
Comment (t0xic0der) [inline CONTRIBUTING.md:36]: The monotype is not properly rendered. Please add a ` in the end before the .

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  2. When ready, add the changes with `git add "file name"
```
Comment (t0xic0der) [inline CONTRIBUTING.md:37]: The monotype is not properly rendered. Please add a ` in the end before the .

**Review by t0xic0der — CHANGES_REQUESTED** [review]
There's this inconsistency with the use of fullstops. :smile: You would want to either use it everywhere or refrain from using it at all.

**Comment by ramyaparimi** [comment]
@t0xic0der thanks for reviewing the changes. I updated the document as per suggestions 😀

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -3,17 +3,17 @@ Your interest in contributing to this project is greatly appreciated. Please kno
 **Here are some ways you can help that doesn't involve coding:**
 
 - Review any active PRs. If the PR needs any changes, please exlpain them in a simple way.
```
Comment (t0xic0der) [inline CONTRIBUTING.md:5]: `sed -i s/exlpain/explain/g`

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -3,17 +3,17 @@ Your interest in contributing to this project is greatly appreciated. Please kno
...
-- You can test the project to see if it has any bugs. If you come across a bug that has not been addressed before, please explain the scenario in a clear and consice manner.
+- You can test the project to see if it has any bugs. If you come across a bug that has not been addressed before in the [issues](https://github.com/fedora-infra/mote/issues), please create a new issue to explain the scenario in a clear and concise manner.
 
 **If coding is your choice of contribution:**
```
Comment (t0xic0der) [inline CONTRIBUTING.md:8]: Rephrase this as "If you want to contribute with code". @nasirhm what do you think?

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -26,15 +26,15 @@ b. Create your forked repo:
   3. From your terminal go to the directory where you cloned the upstream repo.
   4. Run the command `git remote add origin <SSH URL>`.
   
-c. Before every contribution:
+c. Before every contribution and creating a PR:
```
Comment (t0xic0der) [inline CONTRIBUTING.md:29]: Makes sense. :+1:

**Review by t0xic0der — CHANGES_REQUESTED** [review]
Great work with the improvements. :smile:

File: CONTRIBUTING.md (hunk 8 above)
Comment (ramyaparimi) [reply CONTRIBUTING.md]: lol not sure how I missed this second time :D
